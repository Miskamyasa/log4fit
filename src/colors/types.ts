import {ColorSchemeName as SchemeName} from "react-native"

export type ColorNames =
  | "text"
  | "viewBackground"
  | "dimmedBackground"
  | "buttonBackground"
  | "buttonText"
  | "screenBackground"
  | "headerTitle"
  | "headerBackground"
  | "loaderColor"
  | "loaderBackground"
  | "overlayBackground"
  | "dividerColor"
  | "alwaysWhite"

export type ColorSchemeName = NonNullable<SchemeName>

export type ThemeProps = {
    light: string
    dark: string
}
