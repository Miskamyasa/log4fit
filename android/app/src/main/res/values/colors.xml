<resources>
  <color name="iconBackground">#FFFFFF</color>
  <color name="colorPrimary">#023c69</color>
  <color name="colorPrimaryDark">#000000</color>
</resources>