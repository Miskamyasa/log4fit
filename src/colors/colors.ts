import {ColorNames, ColorSchemeName} from "./types";


type Colors = Record<ColorSchemeName, Record<ColorNames, string>>;

export const mainBackgroundColor = "#000";

const header = {
  headerBackground: mainBackgroundColor,
  headerTitle: "#fefefe",
};

export const schemes: Colors = Object.freeze({
  light: {
    text: "#333",
    viewBackground: "#fefefe",
    buttonBackground: "#f7f8f9",
    buttonText: "#678",
    screenBackground: "#fefefe",
    loaderBackground: "transparent",
    loaderColor: "#248",
    gotoWorkoutText: "#245",
    gotoWorkoutBackground: "rgba(210, 220, 230, 0.82)",
    overlayBackground: "rgba(210, 220, 230, 0.93)",
    ...header,
  },
  dark: {
    text: "#fefefe",
    viewBackground: mainBackgroundColor,
    buttonBackground: "#070809",
    buttonText: "#f7f8f9",
    screenBackground: mainBackgroundColor,
    loaderBackground: "transparent",
    loaderColor: "#f2f4f8",
    gotoWorkoutText: "#f1f4f9",
    gotoWorkoutBackground: "rgba(0, 0, 0, 0.7)",
    overlayBackground: "rgba(0, 0, 0, 0.89)",
    ...header,
  },
});
