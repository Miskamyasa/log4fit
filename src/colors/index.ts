export * from "./types";
export * from "./colors";
export * from "./useColorScheme";
export * from "./useThemeColor";
