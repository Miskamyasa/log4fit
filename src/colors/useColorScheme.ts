import {useColorScheme as _useColorScheme} from "react-native"

import type {ColorSchemeName} from "./types"

export function useColorScheme(): ColorSchemeName {
    return _useColorScheme() ?? "dark"
}
