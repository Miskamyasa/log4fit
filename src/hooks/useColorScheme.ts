import {ColorSchemeName as SchemeName, useColorScheme as _useColorScheme} from "react-native";

export type ColorSchemeName = NonNullable<SchemeName>;

export function useColorScheme(): ColorSchemeName {
  return _useColorScheme() || "dark";
}
