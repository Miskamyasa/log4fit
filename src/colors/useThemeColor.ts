import {schemes} from "./colors"
import {ThemeProps, ColorNames} from "./types"
import {useColorScheme} from "./useColorScheme"


export function useThemeColor(colorName: ColorNames, props?: ThemeProps): string {
  const colorScheme = useColorScheme()
  return props && props[colorScheme] || schemes[colorScheme][colorName]
}
