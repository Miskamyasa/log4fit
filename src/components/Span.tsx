import {memo, ReactElement, useMemo} from "react"
import {Text, TextStyle} from "react-native"

import {ColorNames, ThemeProps} from "../colors/types"
import {useThemeColor} from "../colors/useThemeColor"


type _Props = {
  colorName?: ColorNames,
  weight?: "400" | "600" | "900",
  size?: number,
  style?: TextStyle,
  children?: string | number | (string | number)[],
  theme?: ThemeProps,
  lines?: number | undefined,
  flex?: boolean,
}

function Span(props: _Props): ReactElement {
  const {
    style,
    colorName = "text",
    size = 14,
    weight = "400",
    lines,
    flex,
    children = "",
  } = props

  const color = useThemeColor(colorName)

  const styles = useMemo(() => {
    const _style: TextStyle = {
      color,
      fontSize: size,
      fontWeight: weight,
    }
    if (flex) {
      _style.flex = 1
    }
    // TODO  â†“  maybe try like that
    // transform: [{skewX: "-2deg"}],
    return [_style, style]
  }, [flex, color, size, weight, style])

  return (
    <Text
      style={styles}
      numberOfLines={lines}
      ellipsizeMode="tail">
      {children}
    </Text>
  )
}

export default memo(Span)
