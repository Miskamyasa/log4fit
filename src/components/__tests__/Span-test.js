import renderer from "react-test-renderer";

import {useColorScheme} from "../../colors/useColorScheme";
import Span from "../Span";


jest.mock("../../colors/useColorScheme");

describe("components: Text", () => {
  const DEFAULT = "light";
  useColorScheme.mockReturnValue(DEFAULT);

  // const {result: colorScheme} = renderHook(() => useColorScheme());

  it("renders with default (light theme)", () => {
    const tree = renderer.create(<Span>Text</Span>).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("renders with dark theme", () => {
    const DARK = "dark";
    useColorScheme.mockReturnValue(DARK);
    const tree = renderer.create(<Span>Text</Span>).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("renders with custom color #999", () => {
    const color = "#999";
    const props = {light: color, dark: color};
    const tree = renderer.create(<Span {...props}>Text</Span>).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("renders with custom styles {fontSize: 12}", () => {
    const props = {style: {fontSize: 12}};
    const tree = renderer.create(<Span {...props}>Text</Span>).toJSON();
    expect(tree).toMatchSnapshot();
  });

});
