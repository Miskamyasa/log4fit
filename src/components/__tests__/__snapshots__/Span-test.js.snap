// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components: Text renders with custom color #999 1`] = `
<Text
  ellipsizeMode="tail"
  style={
    [
      {
        "color": "#f2f3f4",
        "fontSize": 14,
        "fontWeight": "400",
      },
      undefined,
    ]
  }
>
  Text
</Text>
`;

exports[`components: Text renders with custom styles {fontSize: 12} 1`] = `
<Text
  ellipsizeMode="tail"
  style={
    [
      {
        "color": "#f2f3f4",
        "fontSize": 14,
        "fontWeight": "400",
      },
      {
        "fontSize": 12,
      },
    ]
  }
>
  Text
</Text>
`;

exports[`components: Text renders with dark theme 1`] = `
<Text
  ellipsizeMode="tail"
  style={
    [
      {
        "color": "#f2f3f4",
        "fontSize": 14,
        "fontWeight": "400",
      },
      undefined,
    ]
  }
>
  Text
</Text>
`;

exports[`components: Text renders with default (light theme) 1`] = `
<Text
  ellipsizeMode="tail"
  style={
    [
      {
        "color": "#234",
        "fontSize": 14,
        "fontWeight": "400",
      },
      undefined,
    ]
  }
>
  Text
</Text>
`;
